TheWitcher.Business/HeroService.cs

```diff
@@ -60,9 +61,14 @@ private bool ProcessCoefficient(Heroes hero, Quest quest, int heroPower)
                 hero.ReleaseDate.Value.AddSeconds(newQuestTimeInSeconds);
                 return true;
             }
-            catch(Exception e)
+            catch(DivideByZeroException e)
             {
-                return false;
+                throw new DivideByZeroException("Division by zero detected!", e);
```

Comment:
"Division by zero detected!" - useless error message. It should let you know what exactly went wrong. For example you are not able to answer "Which field is 0?" looking only at this message. In future, please provide more meaningful message. For example: "Quest complexity can not be less then 1".